### PR TITLE
sort: :sparkles: add version sort

### PR DESCRIPTION
**Pull request by zwpaper**
fix https://github.com/Peltoche/lsd/issues/394

### CODE REVIEW

File: src/sort.rs [hunk 1]
```diff
@@ -56,6 +58,10 @@ fn by_date(a: &Meta, b: &Meta) -> Ordering {
     b.date.cmp(&a.date).then(a.name.cmp(&b.name))
 }
 
+fn by_version(a: &Meta, b: &Meta) -> Ordering {
+    compare(&a.name.name, &b.name.name).then(compare(&b.name.name, &a.name.name))
```
Comment (meain) [inline src/sort.rs:62]: Any specific reason why you have a `.then` chained onto it?

File: tests/integration.rs [hunk 2]
```diff
@@ -260,6 +260,50 @@ fn test_show_folder_of_symlink_for_long_multi() {
         .stdout(predicate::str::contains("link/:"));
 }
 
+#[test]
+fn test_version_sort() {
```
Comment (meain) [inline tests/integration.rs:264]: I think `sort.rs` would be a better place for this test than in `integration.rs`.
I guess we will keep this test here anyways, should help with an integration test for sort. But could you have a similar test in `sort.rs`?

Comment (meain) [reply tests/integration.rs:264]: Maybe also add one with items like this:
```
0.3.7  0.11.5
```
Probably not necessary, but just to be safe.

File: src/sort.rs (hunk 1 above)
Comment (zwpaper) [reply src/sort.rs]: my bad, this is a mistake. updated

File: tests/integration.rs (hunk 2 above)
Comment (zwpaper) [reply tests/integration.rs:264]: added, IMO, `integration.rs` is where we did the app functional testing, `lsd` as the entry.

of cause, some more package unit tests would always be better.

**Comment by zwpaper** [comment]
I read the comment in ur email, it's a good idea to add some negative case, so I added some more file names as the test.

origin ls:

```
[vagrant@arch ~]$ ls -l -v
total 0
-rw-r--r-- 1 vagrant vagrant   0 Sep 19 09:30 0.2
-rw-r--r-- 1 vagrant vagrant   0 Sep 19 09:30 0.3.7
-rw-r--r-- 1 vagrant vagrant   0 Sep 19 09:30 0.11
-rw-r--r-- 1 vagrant vagrant   0 Sep 19 09:30 0.11.5
-rw-r--r-- 1 vagrant vagrant   0 Sep 19 09:30 1
-rw-r--r-- 1 vagrant vagrant   0 Sep 19 09:30 2
-rw-r--r-- 1 vagrant vagrant   0 Sep 19 09:30 11
-rw-r--r-- 1 vagrant vagrant   0 Sep 19 09:30 11a
-rw-r--r-- 1 vagrant vagrant   0 Sep 19 09:30 22
```

**Pull request merged**